data/abilities.ts

```diff
@@ -1206,6 +1213,9 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
 				}
 			}
 		},
+		onWeatherChange() {
+			(this.effect as AbilityData).onStart?.call(this, this.effectState.target);
+		},
```

Comment:
Can you not use `this.battle.singleEvent('Start', this.effect, this.effectState, this.effectState.target);`?

Also, doesn't `onWeatherChange` take the ability holder as its first parameter i.e. this then becomes
```suggestion
		onWeatherChange(pokemon) {
			this.battle.singleEvent('Start', this.effect, this.effectState, pokemon);
		},
```